src/sst_iter.rs

```diff
@@ -17,8 +17,8 @@ enum FetchTask {
     Finished(VecDeque<Arc<Block>>),
 }
 
-pub(crate) struct SstIterator<'a> {
-    table: &'a SsTableHandle,
+pub(crate) struct SstIterator<'a, H: AsRef<SsTableHandle> = &'a SsTableHandle> {
+    table: H,
```

Comment:
should i add a comment explaining why we are using a Trait bound here instead of the concrete type, i have already added  an explanation to the commit message, should we add it here too?

Reply:
Ya, I think a comment would be great. This is some funky lookin' code, so good to over explain rather than under explain. :)

Reply:
done